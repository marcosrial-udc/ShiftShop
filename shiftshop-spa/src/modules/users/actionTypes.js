export const GET_ROLES_COMPLETED = "project/users/getRolesCompleted";
export const LOGIN_COMPLETED = "project/users/loginCompleted";
export const LOGOUT = "project/users/logout";
export const GET_USERS_COMPLETED = "project/users/getUsersCompleted";
export const CLEAR_USERS_SEARCH = "project/users/clearUsersSearch";

