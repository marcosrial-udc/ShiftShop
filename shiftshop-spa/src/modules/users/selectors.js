const getModuleState = state => state.users;

export const getUser = state =>
    getModuleState(state).user;

export const isLoggedIn = state =>
    getUser(state) !== null;
