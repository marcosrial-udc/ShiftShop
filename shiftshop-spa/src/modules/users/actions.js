import * as actionTypes from './actionTypes';

export const login = (username, password) => ({
    type: actionTypes.LOGIN_COMPLETED,
    authenticatedUser: ({user: {username, password}})
});

export const logout = () => {
    return {type: actionTypes.LOGOUT};
};
