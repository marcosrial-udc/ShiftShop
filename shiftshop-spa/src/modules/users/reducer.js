import {combineReducers} from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
    roles: [],
    user: null,
    usersSearch: null,
    blockedUsersSearch: null
};

const roles = (state = initialState.roles, action) => {

    switch (action.type) {

        case actionTypes.GET_ROLES_COMPLETED:
            return action.roles;

        default:
            return state;
    }

};

const user = (state = initialState.user, action) => {

    switch (action.type) {

        case actionTypes.LOGIN_COMPLETED:
            return action.authenticatedUser.user;

        case actionTypes.LOGOUT:
            return initialState.user;

        default:
            return state;
    }

};

const usersSearch = (state = initialState.usersSearch, action) => {

    switch (action.type) {

        case actionTypes.GET_USERS_COMPLETED:
            return action.usersSearch;

        default:
            return state;
    }

};

const blockedUsersSearch = (state = initialState.blockedUsersSearch, action) => {

    switch (action.type) {

        case actionTypes.GET_BLOCKED_USERS_COMPLETED:
            return action.blockedUsersSearch;

        case actionTypes.CLEAR_BLOCKED_USERS_SEARCH:
            return initialState.blockedUsersSearch;

        default:
            return state;
    }

};


const reducer = combineReducers({
    roles,
    user,
    usersSearch,
    blockedUsersSearch
});

export default reducer;
