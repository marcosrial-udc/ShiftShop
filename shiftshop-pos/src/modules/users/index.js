import * as actionTypes from './actionTypes';
import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as LoginPortal} from './components/LoginPortal';

export default {actionTypes, actions, reducer, selectors};
