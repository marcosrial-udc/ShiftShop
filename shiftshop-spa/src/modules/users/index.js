import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Role} from './constants/Role';
export {default as LoginPortal} from './components/LoginPortal';
export {default as LogOut} from './components/LogOut';
export {default as ProfileMenu} from './components/ProfileMenu';

export default {actions, actionTypes, reducer, selectors};
