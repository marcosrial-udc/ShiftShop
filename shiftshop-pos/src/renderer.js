import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {Provider} from 'react-redux';
import {MuiThemeProvider} from '@material-ui/core';
import {SnackbarProvider} from 'notistack';
import {IntlProvider} from 'react-intl';

/* Async/await regenerator necesaries */
import 'core-js/stable';
import 'regenerator-runtime/runtime';

import './assets/scss/index.scss';
import theme from './theme'

import {App} from './modules/app';

import configureStore from './store';
import {initReactIntl} from "./i18n";
import './polyfills';

/* Configure store. */
const store = configureStore();

/* Configure i18n. */
const {locale, messages} = initReactIntl();

ReactDOM.render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <IntlProvider locale={locale} messages={messages}>
          <SnackbarProvider anchorOrigin={{vertical: 'top', horizontal: 'center'}}
                            autoHideDuration={2500}>
            <App/>
          </SnackbarProvider>
        </IntlProvider>
      </Provider>
    </MuiThemeProvider>,
    document.getElementById("root"));
