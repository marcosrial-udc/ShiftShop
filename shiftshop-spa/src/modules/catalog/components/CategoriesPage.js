import React from 'react';
import {useSelector} from 'react-redux';
import {Grid, Typography} from '@material-ui/core';
import {FormattedMessage} from 'react-intl';

import useStyles from '../styles/CategoriesPage';

import AddCategory from './AddCategory';

import users, {Role} from '../../users';

const CategoriesPage = () => {
    const classes = useStyles();

    const user = useSelector(users.selectors.getUser);

    return (
        <div>
            <Grid className={classes.header} alignItems="flex-end" container justify="space-between" spacing={3}>
                <Grid item>
                    <Typography variant="overline">
                        <FormattedMessage id="project.app.nav.catalog"/>
                    </Typography>
                    <Typography variant="h3">
                        <FormattedMessage id="project.catalog.CategoriesPage.title"/>
                    </Typography>
                </Grid>
                {users.selectors.hasRole(user, [Role.ADMIN]) &&
                    <Grid item>
                        <AddCategory/>
                    </Grid>}
            </Grid>
            <Grid className={classes.grid} container spacing={2}>
                {/* Page content */}
            </Grid>
        </div>
    );

};

export default CategoriesPage;
