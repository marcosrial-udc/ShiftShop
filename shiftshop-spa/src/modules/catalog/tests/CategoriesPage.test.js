import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl';

import {initReactIntl} from "../../../i18n";

import CategoriesPage from '../components/CategoriesPage';
import {Role} from '../../users';

jest.mock('../components/AddCategory', () => () => (<mockAddCategory/>));
jest.mock('../components/CategoriesList', () => () => (<mockCategoriesList/>));

const {messages} = initReactIntl();

describe('CategoriesPage snapshot', () => {

    const middlewares = [];
    const mockStore = configureStore(middlewares);

    const createCategoriesPage = store => {

        return renderer.create(
            <Provider store={store}>
                <IntlProvider locale='en' messages={messages}>
                    <CategoriesPage/>
                </IntlProvider>
            </Provider>
        );

    };

    test('Without addCategory', () => {

        // Initialize mockstore with empty state
        const initialState = {users: {user: {username: 'user', roles: [Role.SALESMAN]}}};
        const store = mockStore(initialState);

        const categoriesPage = createCategoriesPage(store);

        expect(categoriesPage.toJSON()).toMatchSnapshot();

    });

    test('With addCategory', () => {

        // Initialize mockstore with empty state
        const initialState = {users: {user: {username: 'user', roles: [Role.ADMIN]}}};
        const store = mockStore(initialState);

        const categoriesPage = createCategoriesPage(store);

        expect(categoriesPage.toJSON()).toMatchSnapshot();

    });

});
