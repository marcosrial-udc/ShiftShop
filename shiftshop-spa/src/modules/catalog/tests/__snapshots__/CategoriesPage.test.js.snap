// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPage snapshot With addCategory 1`] = `
<div>
  <div
    className="MuiGrid-root makeStyles-header-2 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="MuiTypography-root MuiTypography-overline"
      >
        Catalog
      </span>
      <h3
        className="MuiTypography-root MuiTypography-h3"
      >
        Categories
      </h3>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <mockAddCategory />
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-grid-1 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <mockCategoriesList />
    </div>
  </div>
</div>
`;

exports[`CategoriesPage snapshot Without addCategory 1`] = `
<div>
  <div
    className="MuiGrid-root makeStyles-header-2 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="MuiTypography-root MuiTypography-overline"
      >
        Catalog
      </span>
      <h3
        className="MuiTypography-root MuiTypography-h3"
      >
        Categories
      </h3>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-grid-1 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <mockCategoriesList />
    </div>
  </div>
</div>
`;
